packages/devtools_app/lib/src/screens/memory/panes/allocation_tracing/allocation_profile_tracing_tree.dart

```diff
@@ -145,11 +200,24 @@ class _AllocationProfileTracingTreeHeader extends StatelessWidget {
       tall: true,
       needsTopBorder: false,
       actions: [
+        const Spacer(),
+        TabBar(
+          labelColor:
+              textTheme.bodyText1?.color ?? colorScheme.defaultForeground,
+          tabs: tabs,
+          isScrollable: true,
+          controller: tabController,
+        ),
+        const SizedBox(width: denseSpacing),
         ExpandAllButton(
           onPressed: () => updateTreeStateCallback(
             () {
+              final isBottomUp = tabs[tabController.index] ==
+                  AllocationTracingTree._bottomUpTab;
               final data = controller.selectedTracedClassAllocationData!;
-              for (final root in data.bottomUpRoots) {
+              final roots =
+                  isBottomUp ? data.bottomUpRoots : data.callTreeRoots;
+              for (final root in roots) {
                 root.expandCascading();
```

Comment:
can you remind me again why we are expanding all the roots? Expanding all call tree roots could prevent the user from seeing more than the first root's call path.

Reply:
We don't do this by default, only when the user presses the `Expand All` button.